servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/ByteToMessageDecoder.java

```diff
@@ -242,7 +242,7 @@ protected ByteBuf swapAndCopyCumulation(final ByteBuf cumulation, final ByteBuf
     protected void cumulationReset() {
     }
 
-    private void releaseCumulation() {
+    protected final void releaseCumulation() {
```

Comment:
It may have some data in the `cumulation` buffer. On `handlerRemoved` event it will fire those pending data through the pipeline. By releasing it manually, we can avoid seeing more data.

Reply:
@Scottmitch you were right, when we do `replace`, netty first adds a new handler after the current one and then removes the existing handler. Therefore, even if we had something in the accumulation buffer, it will be discarded by the new handler. So, we can avoid exposing `releaseCumulation()` 🎉